internal/git/git.go

```diff
@@ -339,8 +339,75 @@ func (g *GitOperations) runCmd(ctx context.Context, directory string, args ...st
 
 	if err := cmd.Wait(); err != nil {
 		// exitErr := err.(*exec.ExitError)
-		return "", stderrBuf.String(), err
+		return stdoutBuf.String(), stderrBuf.String(), err
 	}
 
 	return stdoutBuf.String(), stderrBuf.String(), nil
 }
+
+// replace lines 348 to 386
+func (g *GitOperations) HasConflict(ctx context.Context, proposedBranch, activeBranch string) (bool, error) {
+	logger := log.FromContext(ctx)
+	repoPath := gitpaths.Get(g.gap.GetGitHttpsRepoUrl(*g.gitRepo) + g.pathContext)
+
+	// Fetch both branches from origin
+	if _, stderr, err := g.runCmd(ctx, repoPath, "fetch", "origin", activeBranch, proposedBranch); err != nil {
```

Comment:
Is this going to double-up our number of fetches, i.e. network round trips? Can we instead assume that this has already been done?